lib/recipe.rb

```diff
@@ -183,7 +183,7 @@
         "client_name"         => node.name,
         "daemon_mode"					=> daemon_mode,
         "data_collector_url"  => Chef::Config[:data_collector][:server_url],
-        "entity_uuid"         => Chef::JSONCompat.parse(Chef::FileCache.load("data_collector_metadata.json"))["node_uuid"],
+        "entity_uuid"         => Chef::DataCollector::Messages.node_uuid,
```

Comment:
Chef-15 breaks this again, can you use `Chef::Config[:chef_guid]` here instead?  that will be backcompat to this commit:  https://github.com/chef/chef/pull/7016/files

You could also probably do `Chef::Config[:chef_guid] || Chef::DataCollector::Messages.node_uuid` maybe with a `defined?(Chef::DataCollector::Messages)` guard.

Reply:
imma submit a PR..